src/index.js

```diff
@@ -3,247 +3,277 @@
   Author Tobias Koppers @sokra
 */
 
-// TODO: temporarily disabled rules
-/* eslint-disable
-  no-undefined,
-  no-param-reassign,
-  no-underscore-dangle,
-  import/order
-*/
 import { SourceMapConsumer } from 'source-map';
-import { ConcatSource, RawSource, SourceMapSource } from 'webpack-sources';
-
+import { SourceMapSource, RawSource, ConcatSource } from 'webpack-sources';
 import RequestShortener from 'webpack/lib/RequestShortener';
 import ModuleFilenameHelpers from 'webpack/lib/ModuleFilenameHelpers';
+import uglify from 'uglify-es';
 
-import uglify from 'uglify-js';
+/* eslint-disable
+  no-param-reassign
+*/
+
+const warningRegex = /\[.+:([0-9]+),([0-9]+)\]/;
+
+const defaultUglifyOptions = {
+  output: {
+    comments: /^\**!|@preserve|@license|@cc_on/,
+    beautify: false,
+    semicolons: true,
+    shebang: true,
+  },
+};
 
 class UglifyJsPlugin {
   constructor(options) {
     if (typeof options !== 'object' || Array.isArray(options)) {
-      options = {};
+      this.options = {};
+    } else {
+      this.options = options || {};
     }
-    if (typeof options.compressor !== 'undefined') {
-      options.compress = options.compressor;
+
+    this.options.test = this.options.test || /\.js($|\?)/i;
+    this.options.warningsFilter = this.options.warningsFilter || (() => true);
+
+    this.uglifyOptions = this.options.uglifyOptions || {};
```

Comment:
i actually like @Hartorn approach here because its more clear of an indicator *hey... these are options sent straight to uglify"